src/python/zquantum/core/circuits/_gates.py

```diff
@@ -353,9 +353,12 @@ def params_ordering(self) -> Tuple[Parameter, ...]:
         return self.gate_definition.params_ordering
 
     def __call__(self, *gate_params):
-        return self.matrix.subs(
-            {symbol: arg for symbol, arg in zip(self.params_ordering, gate_params)}
-        )
+        if isinstance(self.matrix, sympy.Matrix):
+            return self.matrix.subs(
+                {symbol: arg for symbol, arg in zip(self.params_ordering, gate_params)}
+            )
+        else:
+            return self.matrix
```

Comment:
Line 361 is not covered. I suggest adding a test case for using `CustomGateMatrixFactory` with numpy array in place of sympy matrix.